src/test/java/HW17.java

```diff
@@ -0,0 +1,22 @@
+
+import org.testng.annotations.Test;
+
+import java.time.Duration;
+
+
+public class HW17 extends MyBaseTest {
+
+
+ @Test
+ public void addASongToPlaylist(){
+
+  navigateToPage();
+  login("zahinedu@gmail.com","te$t$tudent");
+  searchSong("take my hand");
+  clickviewAll ();
+  addField ();
+  addToField();
+  verifyAddField ();
+  songField();
```

Comment:
**I would do it in another way, but your solution is also fine !**

For example

**Use in test methods, the following logical split:   GIVEN – WHEN - THEN**

```
 @Test
    public void testChangeProfileName (){
        // GIVEN
        String newName  = getRandomString();
        String currentPassword = "te$t$tudent";
        provideEmail("demo@class.com");
        providePassword();
        clickSubmitBtn();

         // WHEN
        openUserProfilePage();
        setName(newName);
        setCurrentPassword(currentPassword);

         // THEN
        Assert.assertTrue(getSuccessPopUp().isDisplayed());
        Assert.assertEquals(newName, getUsersNameOnHeader());
        Assert.assertEquals(newName, getUsernameProfileField());